x/stakeibc/keeper/msg_server_test.go

```diff
@@ -2575,3 +2575,48 @@ func (s *KeeperTestSuite) TestResumeHostZone_UnhaltedZones() {
 	expectedErrorMsg := fmt.Sprintf("invalid chain id, zone for %s not halted: host zone is not halted", HostChainId)
 	s.Require().Equal(expectedErrorMsg, err.Error(), "should return correct error msg")
 }
+
+// ----------------------------------------------------
+//	           SetCommunityPoolRebate
+// ----------------------------------------------------
+
+func (s *KeeperTestSuite) TestSetCommunityPoolRebate() {
+	rebateInfo := types.CommunityPoolRebate{
+		LiquidStakeAmount: sdk.NewInt(1000),
+		RebatePercentage:  sdk.MustNewDecFromStr("0.5"),
+	}
+
+	// Set host zone with no rebate
+	hostZone := types.HostZone{
+		ChainId: HostChainId,
+	}
+	s.App.StakeibcKeeper.SetHostZone(s.Ctx, hostZone)
+
+	// Submit a message to create the rebate
+	msg := types.MsgSetCommunityPoolRebate{
+		ChainId:            HostChainId,
+		RebatePercentage:   rebateInfo.RebatePercentage,
+		LiquidStakedAmount: rebateInfo.LiquidStakeAmount,
+	}
+	_, err := s.GetMsgServer().SetCommunityPoolRebate(s.Ctx, &msg)
+	s.Require().NoError(err, "no error expected when registering rebate")
+
+	// Confirm the rebate was updated
+	actualHostZone := s.MustGetHostZone(HostChainId)
+	s.Require().Equal(rebateInfo, *actualHostZone.CommunityPoolRebate, "rebate")
```

Comment:
nit: should this error be a bit more descriptive? Maybe something like "rebate able to be read on host zone"?

Reply:
Amazing, resolving!